ext/drmgr/drmgr.c

```diff
@@ -3145,6 +3145,7 @@ drmgr_emulation_init(void)
 static ptr_int_t
 get_emul_note_val(int enote_val)
 {
+    ASSERT(note_base_emul > 0, "note_base_emul not initialized");
```

Comment:
How do you know 0 is not a valid value?

We already have an assert in drmgr_emulation_init().  If we really want to add a check here, let's use the same symbolic value instead of 0.  I would also suggest moving such a check to the high-level calls instead of a low-level called-multiple-times function.

Reply:
> How do you know 0 is not a valid value?

`DRMGR_NOTE_NONE` is zero, also `drmgr_reserve_note_range` uses `DRMGR_NOTE_NONE + 1` as the first free range start: https://github.com/DynamoRIO/dynamorio/blob/b39b85d195b63701e17847b849f4fa4d5c37cce4/ext/drmgr/drmgr.c#L3086

>  I would also suggest moving such a check to the high-level calls instead of a low-level called-multiple-times function.

The get_note_val functions seemed like the right place for the check; it may be out of place in other functions that call get_note_val (like drmgr_insert_emulation_start, drx_restore_arith_flags, etc). If we're worried about performance overhead, then I'm fine with just removing this additional check. I guess I was being paranoid anyway.

Reply:
> The get_note_val functions seemed like the right place for the check; it may be out of place in other functions 

I assumed the worry is a user incorrectly not calling dmgr_init during process init.  If that's the worry, then checking that init has been called in the API routines the user calls, like drmgr_insert_emulation_start, seems like the place to put the checks.  That was my logic.

Reply:
Sure. Can add some checks as part of #4885.